cmd/demo/demo_test.go

```diff
@@ -24,72 +21,61 @@ var (
 	timeout = time.Second * 30
 )
 
+type Cmd struct {
+	*exec.Cmd
+	stdin io.WriteCloser
+}
+
+func nodeCmd(name string) (*Cmd, error) {
+	args := []string{
+		"run", "../../main.go",
+		"demo",
+		"--config", fmt.Sprintf("../../%v.yaml", name),
+		"--network", "../../network.yaml",
+		"--log-level", "trace",
+		"--log-file", fmt.Sprintf("%v.log", name),
+		"--stdio",
+	}
+	cmd := exec.Command("go", args...)
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return nil, err
+	}
+	return &Cmd{cmd, stdin}, nil
+}
+
 func TestNodes(t *testing.T) {
-	alice, _, err := expect.Spawn("go run ../../main.go demo --config ../../alice.yaml --network ../../network.yaml --log-level trace --test-api true --log-file alice.log", -1)
+	// Start Alice
+	alice, err := nodeCmd("alice")
 	require.NoError(t, err)
-	defer alice.Close()
-	time.Sleep(time.Second * 2)
+	require.NoError(t, alice.Start())
+	defer alice.Process.Kill()
+	time.Sleep(5 * time.Second)
 
-	bob, _, err := expect.Spawn("go run ../../main.go demo --config ../../bob.yaml --network ../../network.yaml --log-level trace --log-file bob.log", -1)
+	// Start Bob
+	bob, err := nodeCmd("bob")
 	require.NoError(t, err)
-	defer bob.Close()
-
-	// Alice start
-	_, _, e := alice.Expect(any, timeout)
-	require.NoError(t, e)
-
-	// Bob start
-	_, _, e = bob.Expect(any, timeout)
-	require.NoError(t, e)
-	time.Sleep(time.Second * 5)
+	require.NoError(t, bob.Start())
+	defer bob.Process.Kill()
+	time.Sleep(5 * time.Second)
 
-	// Alice connect to Bob
-	require.NoError(t, sendSynchron(t, alice, "connect bob\n"), "connecting")
-	t.Log("Alice connected")
-	time.Sleep(time.Second * 1)
-	// Alice open channel to Bob
-	require.NoError(t, sendSynchron(t, alice, "open bob 1000 1000\n"), "opening")
+	// Alice opens channel with Bob
+	require.NoError(t, sendCommand(alice, "open bob 1000 1000\n"), "proposing channel")
+	require.NoError(t, sendCommand(bob, "y\n"), "accepting channel proposal")
 	t.Log("Opening channel…")
 	time.Sleep(time.Second * 5)
-	// Alice send to Bob and Bob to Alice
+
+	// Alice sends to Bob and Bob to Alice
 	for i := 0; i < 25; i++ {
 		t.Log("Sending payment… (alice->bob)")
-		require.NoError(t, sendSynchron(t, alice, "send bob 1\n"))
+		require.NoError(t, sendCommand(alice, "send bob 1\n"))
 		t.Log("Sending payment… (bob->alice)")
-		require.NoError(t, sendSynchron(t, bob, "send alice 2\n"))
+		require.NoError(t, sendCommand(bob, "send alice 2\n"))
 	}
 	t.Log("Done")
```

Comment:
We are sadly still missing on-chain balance assertions after the channel settling (also missing).  
I could add an issue for that, since i dont think it would be trivial to do here.

Reply:
yes, please add an issue.

Reply:
Issue is here https://github.com/perun-network/perun-eth-demo/issues/28